lib/rucio/api/replica.py

```diff
@@ -107,20 +108,24 @@ def declare_bad_file_replicas(replicas, reason, issuer, vo='def', session=None):
                 "rse_id": rse_id,
                 "name": r["name"]
             })
-    undeclared = replica.declare_bad_file_replicas(replicas_lst, reason=reason, issuer=issuer, status=BadFilesStatus.BAD, session=session)
+
+    rse_id_to_name = invert_dict(rse_map)   # RSE id -> RSE name
+
+    undeclared = replica.declare_bad_file_replicas(replicas_lst, reason=reason, issuer=issuer, status=BadFilesStatus.BAD,
+                                                   force=force, session=session)
     out = {}
     for rse_id, ulist in undeclared.items():
         if ulist:
             rse_name = None
             if rse_id == 'unknown':
                 rse_name = 'unknown'
+            elif rse_id in rse_id_to_name:
+                rse_name = rse_id_to_name[rse_id]
             else:
                 try:
                     uuid.UUID(rse_id)
                     rse_name = get_rse_name(rse_id=rse_id, session=session)
-                except exception.RSENotFound:
-                    rse_name = str(rse_id)
-                except ValueError:
+                except (ValueError, exception.RSENotFound):
```

Comment:
I do find inconsistent how the `rse_id`s returned from `core.declare_bad_file_replicas` are checked against exceptions, while the `rse` coming as parameter from the function caller are not in L105. Typically, it is desirable to early sanitize inputs, instead of their by-products. Actually, I am not even sure that the L125 try-except block would ever trigger. Please can you address that?